wikimedia/wikipedia-ios
# Add watch / unwatch flow to article and diff toolbars

**Phabricator:** 
https://phabricator.wikimedia.org/T335816

### Notes
This PR connects the watch/unwatch/update expiry actions to the article toolbar and diff toolbar. I would still like to look into (carefully) polishing up how the toast banner looks if possible, and to see if there's a better way to point the iPad popover to the right toolbar location. Otherwise I think this is ready for review. I can address those pieces in a followup PR if this gets merged.

This PR is built off of https://github.com/wikimedia/wikipedia-ios/pull/4560. Do not merge until that PR is merged.

### Test Steps
1. Go to an article while logged out on the Staging scheme. Confirm tapping "Watch" more menu option prompts you to log in.
2. Tap watch again after logging in. Confirm you see a toast.
3. Tap toast. Confirm you see action sheet for changing expiry.
4. Change expiry. Confirm you see a toast indicating the new expiry time.
5. Go into airplane mode and repeat some previous actions. Confirm you see a "No Internet connection" banner.
6. Go to a diff screen and repeat previous steps. Confirm watch / unwatch works as expected.
